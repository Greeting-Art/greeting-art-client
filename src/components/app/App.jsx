import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import ArtGenerator from '../../containers/ArtGenerator';
import ECard from '../../containers/ECard';
import Gallery from '../../containers/Gallery';
import AboutUs from '../../containers/AboutUs';
import Alchemy from '../specialized/Alchemy';
import BDRyan from '../specialized/BDRyan';
import QuackMan from '../specialized/QuackMan';
import Header from '../header/Header';
import './app.css';

export default function App() {
  return (
    <Router>
      <Header />
      <Switch>
        <Route
          path="/"
          exact
          render={(routerProps) => <ArtGenerator {...routerProps} />}
        />
        <Route
          path="/ecard"
          exact
          render={(routerProps) => <ECard {...routerProps} />}
        />
        <Route
          path="/gallery"
          exact
          render={(routerProps) => <Gallery {...routerProps} />}
        />
        <Route
          path="/aboutus"
          exact
          render={(routerProps) => <AboutUs {...routerProps} />}
        />
        <Route
          path="/alchemy"
          exact
          render={(routerProps) => <Alchemy {...routerProps} />}
        />
        <Route
          path="/ryan"
          exact
          render={(routerProps) => <BDRyan {...routerProps} />}
        />
        <Route
          path="/quack"
          exact
          render={(routerProps) => <QuackMan {...routerProps} />}
        />
      </Switch>
    </Router>
  );
}
