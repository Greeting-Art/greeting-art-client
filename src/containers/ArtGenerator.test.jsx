/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import App from '../components/app/App';
//import ArtGenerator from './ArtGenerator';
// import ArtGenerator from './ArtGenerator';
// import { rest } from 'msw';
// import { setupServer } from 'msw/node';

jest.mock('./ArtGenerator', () => () => <div>hey</div>);

// const P5Wrapper = function P5Wrapper(props) {
//   var _this = _super.call(this, props) || this;
//   _this.wrapper = null;
//   _this.state = __assign(__assign({}, props), {
//     canvas: null,
//     wrapper: _this.wrapper,
//   });
//   return _this;
// };

//mock the randomization.js file in simple fashion here;

describe('random art generation container', () => {
  it('displays inputs + an image + buttons', async () => {
    // render(<ArtGenerator />);

    // const screenLoading = await screen.getByText('loading...');
    // expect(screenLoading).toMatchSnapshot();

    const { container } = render(<App />);
    expect(container.textContent).toMatch('hey');
  });
});
