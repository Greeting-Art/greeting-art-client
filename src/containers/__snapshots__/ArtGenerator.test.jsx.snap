// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`random art generation container displays inputs + an image + buttons 1`] = `[Function]`;
